### ISSUE TITLE
Bug: Adding repositories to workspace not clear

### ISSUE THREAD

**Issue by FatumaA** [open-sauced/app#2923]
### Describe the bug

When adding a repository in your workspace, the button says "update workspace" which is the same copy as the button asking users to upgrade to pro to be able to get private workspaces. This is confusing and not clear.

The button should read something like "Save changes".

Here is a screenshot and a video illustrating this:

![Screenshot 2024-03-12 at 13 35 31](https://github.com/open-sauced/app/assets/67555014/7be09d5b-208a-4cbd-bf52-3d1594b01f75)

https://github.com/open-sauced/app/assets/67555014/fa7becf6-3a6f-48dd-8352-4c8ab1fb2fb8

### Steps to reproduce

1. Go to workspaces tab
2. Click add a repository
3. Follow prompts to add a repository
4. See it has been added but not saved
5. See the button to save is the "update workspace" button
6. Note that there is another "update workspace" button that implies a different flow

### Browsers

Chrome

### Additional context (Is this in dev or production?)

localhost

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [X] I agree to follow this project's Contribution Docs

**Comment by bdougie** [comment]
I agree with this being unclear. My first experience, I missed that the update button was on the bottom right. Thanks for opening the issue @FatumaA.

My suggestion would be using different button styles or consider adding something help the update button stand out.

**Comment by FatumaA** [comment]
> > the button says "update workspace" which is the same copy as the button asking users to upgrade to pro to be able to get private workspaces.
> 
> Note: the copy is subtly different
> 
> "Upgrade workspace"
> 
> "Update workspace"
> 
> So I agree, it could be more clear to end users which button does which exactly.

Yeah it's easy to miss the difference. 

I suggest placing the button closer to the top of the box holding the list of repositories or the top of the page?

**Comment by bdougie** [comment]
I'd also suggest a confirmation or alert of unsaved changes.

A few times I've clicked away after adding a few repos to realize I did not hit update.

**Comment by nickytonline** [comment]
This may become more clearer by #2982, but the copy for upgrading might make it stand out more if it says "Upgrade Your Workspace". We could also add an icon to that button @isabensusan, like the sparkle icon.

**Comment by bdougie** [comment]
An additional suggestion is adding a confirm when users leave the page without clicking update.

You can add a repos and then click home without saving.

Another suggestion is adding something visual showing they have "unsaved changes"

**Comment by nickytonline** [comment]
> An additional suggestion is adding a confirm when users leave the page without clicking update.
> 
> You can add a repos and then click home without saving.
> 
> Another suggestion is adding something visual showing they have "unsaved changes"

We can add a dirty check if there are unsaved changes and use [onbeforeunload](https://developer.mozilla.org/en-US/docs/Web/API/Window/beforeunload_event). For some reason when I tried onbeforeunload recently, it wasn't kicking in, but I see it working on MDN in their example, so that's the way to go. We can think on a visual indicator, or literally have a message like "unsaved changes".

For the dirty check, we can start in workspaces, but that'd be good to have site wide.
